src/backend/distributed/sql/udfs/drop_old_time_partitions/latest.sql

```diff
@@ -31,26 +36,33 @@ BEGIN
     AND attnum = partition_column_index;
 
     -- we currently only support partitioning by date, timestamp, and timestamptz
+    custom_cast = '';
     IF partition_column_type <> 'date'::regtype
     AND partition_column_type <> 'timestamp'::regtype
     AND partition_column_type <> 'timestamptz'::regtype  THEN
+      SELECT EXISTS(SELECT OID FROM pg_cast WHERE castsource = partition_column_type AND casttarget = 'timestamptz'::regtype) AND
```

Comment:
seeing unqualified names, I'm kind of wondering whether we should SET search_path at the top of this function.

anyway, it would be good to use pg_catalog